frontend/ui/widget/touchmenu.lua

```diff
@@ -992,260 +989,266 @@ function TouchMenu:onBack()
     self:backToUpperMenu()
 end
 
------- the menu search functionality
+-- Menu search feature
 function TouchMenu:search(search_for)
     local found_menu_items = {}
 
-    local MAX_MENU_DEPTH = 20 -- currently our menu needs at least 12 here
-    local function recurse(val, path, text, icon, depth)
+    local MAX_MENU_DEPTH = 10 -- our menu max depth is currently 6
```

Comment:
Innocent question? Why restrict to 6/10? Did you count every leave?
I had the feeling 12 would be right? If you are sure 6/10 is the maximum, 12/20 would do no harm.
Or did you restrict that as an answer to the massive duplicated results?

Reply:
Your code was counting them twice (once when your recurse() met the menu item, and then recurse() it, where you were adding "sub_item_table." to the path, and then called recurse() again to process the sub_item_table.
Our deepest menu item is in `Screensaver>...>Message Settings>Message position` which makes it 6.